src/ng/compile.js

```diff
@@ -1960,19 +1960,21 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
                                       newIsolateScopeDirective, isolateScope);
         }
         if (elementControllers) {
-          // Initialize bindToController bindings for new/isolate scopes
-          var scopeDirective = newIsolateScopeDirective || newScopeDirective;
-          var bindings;
-          var controllerForBindings;
-          if (scopeDirective && elementControllers[scopeDirective.name]) {
-            bindings = scopeDirective.$$bindings.bindToController;
-            controller = elementControllers[scopeDirective.name];
-
-            if (controller && controller.identifier && bindings) {
-              controllerForBindings = controller;
-              thisLinkFn.$$destroyBindings =
+          for (i in elementControllers) {
+            // Initialize bindToController bindings for new/isolate scopes
+            var scopeDirective = newIsolateScopeDirective || controllerDirectives[i];
```

Comment:
If you try to combine scope: {} with scope: true on the same element you'll get an error earlier: 
```
Multiple directives [first, second] asking for new/isolated scope on: <div first="{{1+1}}" second="{{2+2}}">
```
So it's either one iteration for isolated scope, or multiple for scope: true